cmd/control/install.go

```diff
@@ -113,23 +136,137 @@ func runInstall(image, installType, cloudConfig, device, append string, force, r
 		diskType = "gpt"
 	}
 
+	// Versions before 0.8.0-rc2 use the old calling convention (from the lay-down-os shell script)
+	imageVersion := strings.TrimPrefix(image, "rancher/os:v")
+	if image != imageVersion {
+		log.Infof("user spcified different install image: %s != %s", image, imageVersion)
+		imageVersion = strings.Replace(imageVersion, "-", ".", -1)
+		vArray := strings.Split(imageVersion, ".")
+		v, _ := strconv.ParseFloat(vArray[0]+"."+vArray[1], 32)
+		if v < 0.8 || imageVersion == "0.8.0-rc1" {
+			log.Infof("starting installer container for %s", image)
+			if installType == "generic" ||
+				installType == "syslinux" ||
+				installType == "gptsyslinux" {
+				cmd := exec.Command("system-docker", "run", "--net=host", "--privileged", "--volumes-from=all-volumes",
+					"--entrypoint=/scripts/set-disk-partitions", image, device, diskType)
+				cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr
+				if err := cmd.Run(); err != nil {
+					return err
+				}
+			}
+			cmd := exec.Command("system-docker", "run", "--net=host", "--privileged", "--volumes-from=user-volumes",
+				"--volumes-from=command-volumes", image, "-d", device, "-t", installType, "-c", cloudConfig,
+				"-a", kappend)
+			cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr
+			if err := cmd.Run(); err != nil {
+				return err
+			}
+			return nil
+		}
+	}
+
+	useIso := false
```

Comment:
Got it, thanks.  In my horrible hack, I was blindly mounting and loading the os image since I knew that whatever I packed on my ISO was what I needed to keep the Vagrant users happy and because I was hacking my junk on top of 0.7.1 without all your other nice changes (yet).  LGTM.